src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -2657,17 +2657,21 @@ void mvnw(Register Rd, Register Rm,
     f(sidx<<(int)T, 14, 11), f(1, 10), rf(Vn, 5), rf(Vd, 0);
   }
 
-#define INSN(NAME, op)                                                     \
-  void NAME(Register Rd, FloatRegister Vn, SIMD_RegVariant T, int idx) {   \
-    starti;                                                                \
-    f(0, 31), f(T==D ? 1:0, 30), f(0b001110000, 29, 21);                   \
-    f(((idx<<1)|1)<<(int)T, 20, 16), f(op, 15, 10);                        \
-    rf(Vn, 5), rf(Rd, 0);                                                  \
+  void umov(Register Rd, FloatRegister Vn, SIMD_RegVariant T, int idx) {
+    starti;
+    assert(T != Q, "invalid register variant");
+    f(0, 31), f(T == D ? 1 : 0, 30), f(0b001110000, 29, 21);
+    f(((idx << 1) | 1) << (int)T, 20, 16), f(0b001111, 15, 10);
```

Comment:
```(T == D ? 1 : 0)``` is just ```(T == D)```, isn't it?

Reply:
Thanks for your comment.

Yes, (T == D ? 1 : 0) equals to (T == D). However, I prefer the former one because it's more accurate to pass an integer as the first argument of 'f' function, compared to passing a boolean value. Besides, the former one is widely used by the definitions of other instructions.